tests/storage/moosefs/conftest.py

```diff
@@ -13,15 +14,24 @@ def host_with_moosefs(host):
 
     host.activate_smapi_driver('moosefs')
 
-    yield host
-
-    # teardown
+def disable_moosefs(host):
     host.deactivate_smapi_driver('moosefs')
 
     host.yum_restore_saved_state()
     host.ssh(['rm -f /etc/pki/rpm-gpg/RPM-GPG-KEY-MooseFS'])
     host.ssh(['rm -f /etc/yum.repos.d/MooseFS.repo'])
 
+@pytest.fixture(scope='package')
+def pool_with_moosefs(hostA1):
+    pool = Pool(hostA1.hostname_or_ip)
+    for host in pool.hosts:
+        enable_moosefs(host)
+
+    yield pool
+
+    for host in pool.hosts:
+        disable_moosefs(host)
+
```

Comment:
This fixture will behave poorly if for example the third host has moosefs installed on it for some reason. But it's not worse than in other similar tests, so I think I'll try to fix them all at some point.

Reply:
test_reboot should also probably explicitly start the VM on the master host as this is the only host that was rebooted.